edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.CloudProxy.Test/CloudProxyTest.cs

```diff
@@ -49,43 +58,64 @@ public async Task SendMessageMultipleDevicesTest()
 
-            ICloudProxy cloudProxy1 = await this.GetCloudProxyWithConnectionStringKey("device2ConnStrKey");
+            string device2ConnectionString = await SecretsHelper.GetSecretFromConfigKey(Device2ConnStrKey);
+            string device3ConnectionString = await SecretsHelper.GetSecretFromConfigKey(Device3ConnStrKey);
+            string deviceId2 = IotHubConnectionStringBuilder.Create(device2ConnectionString).DeviceId;
+            string deviceId3 = IotHubConnectionStringBuilder.Create(device3ConnectionString).DeviceId;
+
+            ICloudProxy cloudProxy2 = await this.GetCloudProxyFromConnectionStringKey(Device2ConnStrKey);
+            ICloudProxy cloudProxy3 = await this.GetCloudProxyFromConnectionStringKey(Device3ConnStrKey);
 
-            ICloudProxy cloudProxy2 = await this.GetCloudProxyWithConnectionStringKey("device3ConnStrKey");
+            Dictionary<string, IList<IMessage>> sentMessagesByDevice = new Dictionary<string, IList<IMessage>>();
+            sentMessagesByDevice.Add(deviceId2, new List<IMessage>());
+            sentMessagesByDevice.Add(deviceId3, new List<IMessage>());
 
             for (int i = 0; i < messages.Count; i = i + 2)
             {
-                await cloudProxy1.SendMessageAsync(messages[i]);
-                await cloudProxy2.SendMessageAsync(messages[i + 1]);
+                IMessage device2Message = messages[i];
+                IMessage device3Message = messages[i + 1];
+
+                await cloudProxy2.SendMessageAsync(device2Message);
+                await cloudProxy3.SendMessageAsync(device3Message);
+
+                sentMessagesByDevice[deviceId2].Add(device2Message);
+                sentMessagesByDevice[deviceId3].Add(device3Message);
             }
 
-            bool disconnectResult = await cloudProxy1.CloseAsync();
+            bool disconnectResult = await cloudProxy2.CloseAsync();
             Assert.True(disconnectResult);
-            disconnectResult = await cloudProxy2.CloseAsync();
+            disconnectResult = await cloudProxy3.CloseAsync();
             Assert.True(disconnectResult);
 
-            await CheckMessageInEventHub(messages, startTime);
+            await CheckMessageInEventHub(sentMessagesByDevice, startTime);
         }
 
         [Fact]
         [TestPriority(403)]
         public async Task SendMessageBatchTest()
         {
+            string device2ConnectionString = await SecretsHelper.GetSecretFromConfigKey(Device2ConnStrKey);
+            string deviceId2 = IotHubConnectionStringBuilder.Create(device2ConnectionString).DeviceId;
+
             IList<IMessage> messages = MessageHelper.GenerateMessages(4);
             DateTime startTime = DateTime.UtcNow.Subtract(ClockSkew);
-            ICloudProxy cloudProxy = await this.GetCloudProxyWithConnectionStringKey("device2ConnStrKey");
+            ICloudProxy cloudProxy = await this.GetCloudProxyFromConnectionStringKey("device2ConnStrKey");
```

Comment:
Do you want to use constant here? more usages below.
```suggestion
            ICloudProxy cloudProxy = await this.GetCloudProxyFromConnectionStringKey(Device2ConnStrKey);
```